diff --git a/src/app/features/settings/developer-tools/DevelopTools.tsx b/src/app/features/settings/developer-tools/DevelopTools.tsx
index 01e6e6a4..37c6be1e 100644
--- a/src/app/features/settings/developer-tools/DevelopTools.tsx
+++ b/src/app/features/settings/developer-tools/DevelopTools.tsx
@@ -48,6 +48,7 @@ export function DeveloperTools({ requestClose }: DeveloperToolsProps) {
   const [extendedProfile, refreshExtendedProfile] = useExtendedProfile(userId);
 
   const [developerTools, setDeveloperTools] = useSetting(settingsAtom, 'developerTools');
+  const [issueTracker, setIssueTracker] = useSetting(settingsAtom, 'issueTracker');
   const [page, setPage] = useState<DeveloperToolsPage>({ name: 'index' });
   const [globalExpand, setGlobalExpand] = useState(false);
   const [profileExpand, setProfileExpand] = useState(false);
@@ -132,6 +133,27 @@ export function DeveloperTools({ requestClose }: DeveloperToolsProps) {
             <Scroll hideTrack visibility="Hover">
               <PageContent>
                 <Box direction="Column" gap="700">
+                  <Box direction="Column" gap="100">
+                    <Text size="L400">Experimental Features</Text>
+                    <SequenceCard
+                      className={SequenceCardStyle}
+                      variant="SurfaceVariant"
+                      direction="Column"
+                      gap="400"
+                    >
+                      <SettingTile
+                        title="Issue Tracker"
+                        description="Enable per-room issue tracker (experimental)."
+                        after={
+                          <Switch
+                            variant="Primary"
+                            value={issueTracker}
+                            onChange={setIssueTracker}
+                          />
+                        }
+                      />
+                    </SequenceCard>
+                  </Box>
                   <Box direction="Column" gap="100">
                     <Text size="L400">Options</Text>
                     <SequenceCard
diff --git a/src/app/state/settings.ts b/src/app/state/settings.ts
index 95f3ec30..8962168c 100644
--- a/src/app/state/settings.ts
+++ b/src/app/state/settings.ts
@@ -49,6 +49,7 @@ export interface Settings {
   callRingScope: 'dm' | 'nonVoice' | 'all';
   callRingtoneUrl: string | null;
 
+  issueTracker: boolean;
   hour24Clock: boolean;
   dateFormatString: string;
 
@@ -85,6 +86,7 @@ const defaultSettings: Settings = {
 
   showNotifications: true,
   isNotificationSounds: true,
+  issueTracker: false,
   inRoomActivitySound: true,
   inboxUnreadNotifications: false,
   inboxNotifBatchDelay: 60,
